From 7f7c4f7224ebb3773eb515c0e45b1716dc8b49d4 Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf.samuelsson@atmel.com>
Date: Sat, 16 Apr 2011 02:22:16 +0200
Subject: [PATCH 18/24] sound: atmel ac97: add warm reset if no reset pin

In case of no reset pin defined in ac97c driver, we issue a warm reset anyway.

Ported to linux-2.6.38
Signed-off-by: Ulf Samuelsson <ulf.samuelsson@atmel.com>
---
 sound/atmel/ac97c.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/sound/atmel/ac97c.c b/sound/atmel/ac97c.c
index 104f588..81f3fbb 100644
--- a/sound/atmel/ac97c.c
+++ b/sound/atmel/ac97c.c
@@ -1,3 +1,5 @@
+#define DEBUG 12
+#define VERBOSE_DEBUG 12
 /*
  * Driver for Atmel AC97C
  *
@@ -899,6 +901,10 @@ static void atmel_ac97c_reset(struct atmel_ac97c *chip)
 		/* AC97 v2.2 specifications says minimum 1 us. */
 		udelay(2);
 		gpio_set_value(chip->reset_pin, 1);
+	} else {
+		ac97c_writel(chip, MR, AC97C_MR_WRST | AC97C_MR_ENA);
+		udelay(2);
+		ac97c_writel(chip, MR, AC97C_MR_ENA);
 	}
 }
 
-- 
1.6.3.3

